code/game/gamemodes/cult/cult_structures.dm

```diff
@@ -26,17 +39,238 @@
-/*
-/obj/structure/cult/pillar
-	name = "Pillar"
-	desc = "This should not exist"
-	icon_state = "pillar"
-	icon = 'magic_pillar.dmi'
-*/
+
+//Cult versions cuase fuck map conflicts
+/obj/structure/cult/examine(mob/user)
+	. = ..()
+	if(iscultist(user) && cooldowntime > world.time)
+		to_chat(user, "<span class='cultitalic'>The magic in [src] is weak, it will be ready to use again in [getETA()].</span>")
+	to_chat(user, "<span class='notice'>\The [src] is [anchored ? "":"not "]secured to the floor.</span>")
+
+/obj/structure/cult/attackby(obj/I, mob/user, params)
+	if(istype(I, /obj/item/weapon/tome) && iscultist(user))
+		anchored = !anchored
+		to_chat(user, "<span class='notice'>You [anchored ? "":"un"]secure \the [src] [anchored ? "to":"from"] the floor.</span>")
+		playsound(loc, 'sound/hallucinations/wail.ogg', 75, 1)
+		if(!anchored)
+			icon_state = "[initial(icon_state)]_off"
+		else
+			icon_state = initial(icon_state)
+	else
+		return ..()
+
+
+/obj/structure/cult/proc/getETA()
+	var/time = (cooldowntime - world.time)/600
+	var/eta = "[round(time, 1)] minutes"
+	if(time <= 1)
+		time = (cooldowntime - world.time)*0.1
+		eta = "[round(time, 1)] seconds"
+	return eta
+
+/obj/structure/cult/culttalisman
+	name = "altar"
+	desc = "A bloodstained altar dedicated to a cult."
+	icon_state = "talismanaltar"
+	health = 150 //Sturdy
+	death_message = "<span class='warning'>The altar breaks into splinters, releasing a cascade of spirits into the air!</span>"
+	death_sound = 'sound/effects/altar_break.ogg'
+
+/obj/structure/cult/culttalisman/attack_hand(mob/living/user)
+	if(!iscultist(user))
+		to_chat(user, "<span class='warning'>You're pretty sure you know exactly what this is used for and you can't seem to touch it.</span>")
+		return
+	if(!anchored)
+		to_chat(user, "<span class='cultitalic'>You need to anchor [src] to the floor with a tome first.</span>")
+		return
+	if(cooldowntime > world.time)
+		to_chat(user, "<span class='cultitalic'>The magic in [src] is weak, it will be ready to use again in [getETA()].</span>")
+		return
+	var/choice = input(user,"You study the rituals on the altar...","Altar") as null|anything in list("Cultist Dagger", "Eldritch Whetstone","Zealot's Blindfold","Flask of Unholy Water")
+	var/pickedtype
+	switch(choice)
+		if("Eldritch Whetstone")
+			pickedtype = /obj/item/weapon/whetstone/cult
+		if("Zealot's Blindfold")
+			pickedtype = /obj/item/clothing/glasses/night/cultblind
+		if("Flask of Unholy Water")
+			pickedtype = /obj/item/weapon/reagent_containers/food/drinks/bottle/unholywater
+		if("Cultist Dagger")
+			pickedtype = /obj/item/weapon/melee/cultblade/dagger
+	if(pickedtype && Adjacent(user) && src && !qdeleted(src) && !user.incapacitated() && cooldowntime <= world.time)
+		cooldowntime = world.time + 2400
+		var/obj/item/N = new pickedtype(get_turf(src))
+		to_chat(user, "<span class='cultitalic'>You kneel before the altar and your faith is rewarded with an [N]!</span>")
```

Comment:
Since it seems a lot of these structures seem to all spawn objects and enter a cooldown, would it not be wise to create a subtype for that, and also make the delay time adjustable - for further tweaking?